homeassistant/components/media_player/pioneer.py

```diff
@@ -93,9 +134,13 @@ def telnet_command(self, command):
             except (ConnectionRefusedError, OSError):
                 _LOGGER.warning("Pioneer %s refused connection", self._name)
                 return
+            _LOGGER.debug("Sending command %s to %s", command, self.name)
             telnet.write(command.encode("ASCII") + b"\r")
             telnet.read_very_eager()  # skip response
             telnet.close()
+            # Give the connection time to close.  HA will poll immediately
+            # after this and will fail if the connection is still open.
+            time.sleep(0.5)
```

Comment:
Sleeping is not allowed in our thread pool. If it was asyncio sleep it would be ok.